ensembl_dbcopy/admin.py

```diff
@@ -194,6 +196,29 @@ def change_view(self, request, object_id, form_url='', extra_context=None):
                 F('end_date').desc(nulls_first=True)).earliest('auto_id')
         return super().change_view(request, object_id, form_url, context)
 
+    def _get_deletable_objects(self, queryset):
+        return queryset.exclude(Q(status='Creating Requests') | Q(status='Processing Requests'))
+
+    def get_deleted_objects(self, queryset, request):
+        deletable_queryset = self._get_deletable_objects(queryset)
+        return super().get_deleted_objects(deletable_queryset, request)
+
+    def delete_queryset(self, request, queryset):
+        deletable_queryset = self._get_deletable_objects(queryset)
+        count = deletable_queryset.count()
+        super().delete_queryset(request, deletable_queryset)
```

Comment:
delete_queryset returns the number of lines affected by the query: I don't think you need to do the count() and  the `model_ngettext` thing. See https://docs.djangoproject.com/en/2.2/_modules/django/db/models/query/#QuerySet

Reply:
a shame because `queryset.delete()` return  `return deleted, _rows_count` but `delete_queryset` does not return the underlying method returned values! instead of calling parent you could then retrieve the `queryset.delete()` returned values?
See https://github.com/django/django/blob/fed8129276eac973fd48a2f33b7e1735a866628d/django/contrib/admin/options.py#L1106 It's the only thing done by delete_queryset().

Reply:
https://github.com/django/django/blob/e893c0ad8b0b5b0a1e5be3345c287044868effc4/django/db/models/query.py#L715

Reply:
Yeah, good point. That should be fine.